core/src/test/java/kafka/test/junit/ClusterTestExtensions.java

```diff
@@ -162,32 +162,39 @@ private void processClusterTest(ExtensionContext context, ClusterTest annot, Clu
             controllers = annot.controllers();
         }
 
-        if (brokers <= 0 || controllers <= 0) {
-            throw new IllegalArgumentException("Number of brokers/controllers must be greater than zero.");
+        final int disksPerBroker;
+        if (annot.disksPerBroker() == 0) {
+            disksPerBroker = defaults.disksPerBroker();
```

Comment:
Could we leverage the builder instead of creating a bunch of temporary local variables?

Reply:
Yeah, I remove `brokers`, `controllers`, and `disksPerBroker`, but keeping `type`. We can't get attribute from `ClusterConfig#Builder` and we need `type` to run `invocationContexts`.

Reply:
We can leverage ternary conditional operator to make it more simple. For example:
```java
        Type type = annot.clusterType() == Type.DEFAULT ? defaults.clusterType() : annot.clusterType();
        Map<String, String> serverProperties = new HashMap<>();
        for (ClusterConfigProperty property : defaults.serverProperties()) {
            serverProperties.put(property.key(), property.value());
        }
        for (ClusterConfigProperty property : annot.serverProperties()) {
            serverProperties.put(property.key(), property.value());
        }

        ClusterConfig config = ClusterConfig.builder()
            .setType(type)
            .setBrokers(annot.brokers() == 0 ? defaults.brokers() : annot.brokers())
            .setControllers(annot.controllers() == 0 ? defaults.controllers() : annot.controllers())
            .setDisksPerBroker(annot.disksPerBroker() == 0 ? defaults.disksPerBroker() : annot.disksPerBroker())
            .setAutoStart(annot.autoStart() == AutoStart.DEFAULT ? defaults.autoStart() : annot.autoStart() == AutoStart.YES)
            .setName(annot.name().isEmpty() ? null : annot.name())
            .setName(annot.listener().isEmpty() ? null : annot.listener())
            .setServerProperties(serverProperties)
            .setSecurityProtocol(annot.securityProtocol())
            .setMetadataVersion(annot.metadataVersion())
            .build();

        type.invocationContexts(context.getRequiredTestMethod().getName(), config, testInvocations);
```

Reply:
Thanks for the suggestion. It's more clear. Updated it.